tests/test_pipelineLoadSubset.py

```diff
@@ -1,70 +1,138 @@
 import os
 import unittest
 
+from tempfile import NamedTemporaryFile
+from textwrap import dedent
+
 import lsst.pipe.base as pipeBase
 import lsst.utils.tests
 
 
 class PipelineLoadSubsetTest(unittest.TestCase):
+    def setUp(self):
+        frozen_pipeline = dedent("""
+            description: Frozen Pipeline
+            instrument: lsst.obs.subaru.HyperSuprimeCam
+            tasks:
+              isr:
+                class: lsst.ip.isr.IsrTask
+              charImage:
+                class: lsst.pipe.tasks.characterizeImage.CharacterizeImageTask
+              calibrate:
+                class: lsst.pipe.tasks.calibrate.CalibrateTask
+              makeWarpTask:
+                class: lsst.pipe.tasks.makeCoaddTempExp.MakeWarpTask
+                config:
+                - python: config.warpAndPsfMatch.psfMatch.kernel['AL'].alardSigGauss = [1.0, 2.0,
+                    4.5]
+                  matchingKernelSize: 29
+                  makePsfMatched: true
+                  modelPsf.defaultFwhm: 7.7
+                  doApplyExternalPhotoCalib: false
+                  doApplyExternalSkyWcs: false
+                  doApplySkyCorr: false
+                  doWriteEmptyWarps: true
+              assembleCoadd:
+                class: lsst.pipe.tasks.assembleCoadd.CompareWarpAssembleCoaddTask
+              detection:
+                class: lsst.pipe.tasks.multiBand.DetectCoaddSourcesTask
+              mergeDetections:
+                class: lsst.pipe.tasks.mergeDetections.MergeDetectionsTask
+                config:
+                - priorityList:
+                  - i
+                  - r
+              deblend:
+                class: lsst.pipe.tasks.deblendCoaddSourcesPipeline.DeblendCoaddSourcesSingleTask
+              measure:
+                class: lsst.pipe.tasks.multiBand.MeasureMergedCoaddSourcesTask
+              mergeMeasurements:
+                class: lsst.pipe.tasks.mergeMeasurements.MergeMeasurementsTask
+                config:
+                - priorityList:
+                  - i
+                  - r
+              forcedPhotCcd:
+                class: lsst.meas.base.forcedPhotCcd.ForcedPhotCcdTask
+                config:
+                - doApplyExternalPhotoCalib: false
+                  doApplyExternalSkyWcs: false
+                  doApplySkyCorr: false
+              forcedPhotCoadd:
+                class: lsst.meas.base.forcedPhotCoadd.ForcedPhotCoaddTask
+            """)
```

Comment:
What kind of changes are you guarding against by copying things here?  I'm a bit worried that it could just as easily be broken by _not_ getting necessary changes when upstream code changes require them.

If this test isn't actually going to run this pipeline, could it perhaps be a much simpler one that at least looks less like it depends on the real content of concrete PipelineTasks?

Reply:
Things like Eli adding a task to DRP caused this test to fail and break, but really the specifics of the pipeline are not important for this tests. This test does need to include at least enough real tasks that the dependency relations of the tasks could be resolved, and I figured it would make sense in that case to use what tasks we had instead of mocking up yet more things.